stacks/bracketstester.js

```diff
@@ -0,0 +1,30 @@
+const Stack = require('./Stack');
+
+function BracketsTester(stackMaxCapacity) {
+    this.stringStack = new Stack(stackMaxCapacity);
+    this.count = 0;
+}
+
+BracketsTester.prototype.checkBrackets = function(stringValue) {
+
+    Array.from(stringValue).forEach(charValue => this.stringStack.push(charValue));
+
+    while ( this.stringStack.peek() ) {
+
+        var poppedElement = this.stringStack.pop();
+
+        if(this.count === 0 && poppedElement === "(") {//If opening braces comes first -> stop.
```

Comment:
should not need this right because count would not be 0 when you return final result, isnt it?

Reply:
this is for some string like - "Math.min(5,(6-3))(3*4" if last occurrence of bracket is "(" in string then it will return false as it is not closed. first bracket found from stack should be ")".

Reply:
ha but jab pura program khatm bho jaega then wouldnt count be -1

Reply:
there are 3 conditions:
1.) if last bracket occurrence is "(" which means its not closed then first if condition will return false without checking further. Ex. "Math.min(5,(6-3))**(**3*4"

Reply:
There are 3 conditions:
        if(count === 0 && poppedElement === "(") { return false; }
        else if ( poppedElement === ")" ) { count ++; }
        else if ( poppedElement === "(" ) { count --; }
1.) For balanced brackets, ")" will always come last. If last bracket occurrence is "(" which means its not closed then first if condition will return false without checking further. 
Ex. "5*(6-3)(3*4".
2.) If first condition is not the case, count will increment when ")" is found and decrease when ")" is found. So, when the program ends..... count will be zero if brackets are balanced.